service/api_logic/user_logic.py

```diff
@@ -1,26 +1,33 @@
 from flask import current_app, url_for, jsonify
 from flask_mail import Message
 from itsdangerous import URLSafeTimedSerializer
+from dto.api_input import InputUserLogInDTO
 from dto.api_output import OutputUser, OutputLogin
 from database.models import User
 from dto.common_response import CommonResponse, CommonResponseWithUser
-from exept.exeptions import UserDoesNotExistError, IncorrectUsernameOrEmailError, UserAlreadyExistError, IncorrectPasswordError
+from exept.exeptions import UserDoesNotExistError, UserAlreadyExistError
 import bcrypt
 from logger.logger import Logger
 from jinja2 import Environment, FileSystemLoader
 import os
-from flask_jwt_extended import create_access_token, set_access_cookies
+from flask_jwt_extended import create_access_token,create_refresh_token, set_access_cookies, set_refresh_cookies
+from service.api_logic.auth_strategy import AuthManager
 
 
 class UserService:
+
     def __init__(self, user_dal):
         self._user_dal = user_dal
         self._serializer = URLSafeTimedSerializer(current_app.secret_key)
         self._logger = Logger("logger", "all.log").logger
 
 
+    def get_user_by_email_or_username(self, email_front=None, username_front=None):
+        return self._user_dal.get_user_by_email_or_username(email_front, username_front)
+
+
     async def create_user(self, email_front, username_front, password_front):
-        existing_user = self._user_dal.get_user_by_email_or_username(email_front, username_front)
+        existing_user = self.get_user_by_email_or_username(email_front, username_front)
```

Comment:
MINOR:

Why don't we write a separate query to DB which will just check whether a user exists? That way we can spare some memory.